pkg/api/pod/warnings.go

```diff
@@ -234,19 +234,15 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 		warnings = append(warnings, fmt.Sprintf("%s: fractional byte value %q is invalid, must be an integer", fieldPath.Child("spec", "overhead").Key(string(api.ResourceEphemeralStorage)), value.String()))
 	}
 
-	// use of pod seccomp annotation without accompanying field
-	if podSpec.SecurityContext == nil || podSpec.SecurityContext.SeccompProfile == nil {
-		if _, exists := meta.Annotations[api.SeccompPodAnnotationKey]; exists {
-			warnings = append(warnings, fmt.Sprintf(`%s: deprecated since v1.19, non-functional in v1.25+; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompPodAnnotationKey)))
-		}
+	// use of pod seccomp annotation
+	if _, exists := meta.Annotations[api.SeccompPodAnnotationKey]; exists {
+		warnings = append(warnings, fmt.Sprintf(`%s: deprecated since v1.19, non-functional in a future release; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompPodAnnotationKey)))
 	}
 
 	pods.VisitContainersWithPath(podSpec, fieldPath.Child("spec"), func(c *api.Container, p *field.Path) bool {
-		// use of container seccomp annotation without accompanying field
-		if c.SecurityContext == nil || c.SecurityContext.SeccompProfile == nil {
```

Comment:
same comment here about not needing to warn if they're setting both the annotation and the field